gusto/forcing.py

```diff
@@ -26,106 +28,132 @@ def __init__(self, state, euler_poincare=True, linear=False, extra_terms=None):
             warning('Setting euler_poincare to False because you have set linear=True')
         else:
             self.euler_poincare = euler_poincare
+
+        # set up functions
+        self.Vu = state.spaces("HDiv")
+        self.x0 = Function(state.W)   # copy x to here
+        self.test = TestFunction(self.Vu)
+        self.trial = TrialFunction(self.Vu)
+        self.uF = Function(self.Vu)
```

Comment:
Possibly, but more descriptive variable names obviate the need for such comments